src/pg/connection.cr

```diff
@@ -37,6 +37,16 @@ module PG
       end
     end
 
+    # `#initialize` Connect to the server with values of Hash.
+    #
+    # ```
+    # info = { host: "localhost", user: "postgres", password: "password", db_name: "test_db", port: "5432"}
+    # PG::Connection.new(info)
+    # ```
+    def initialize(parameters : Hash(YAML::Type, YAML::Type))
```

Comment:
Following on the discussion, are there any situations where the key/param side isn't just a `String`, so `Hash(String, YAML::Type)`